src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -312,38 +482,55 @@ void DoorShutter_SetupType(DoorShutter* this, PlayState* play) {
     }
 }
 
-f32 func_80996840(PlayState* play, DoorShutter* this, f32 arg2, f32 arg3, f32 arg4) {
+/**
+ * Compute the distance forwards to the player
+ * (perpendicularly to the door's left-right, and negative if the player is behind),
+ * or returns the max float value if the player is too far left/right (`maxDistSides`) or above/below (`maxDistY`).
+ */
+f32 DoorShutter_GetPlayerDistance(PlayState* play, DoorShutter* this, f32 offsetY, f32 maxDistSides, f32 maxDistY) {
     s32 pad;
-    Vec3f sp28;
-    Vec3f sp1C;
+    Vec3f playerPos;
+    Vec3f relPlayerPos;
     Player* player = GET_PLAYER(play);
 
-    sp28.x = player->actor.world.pos.x;
-    sp28.y = player->actor.world.pos.y + arg2;
-    sp28.z = player->actor.world.pos.z;
-    func_8002DBD0(&this->dyna.actor, &sp1C, &sp28);
-    if (arg3 < fabsf(sp1C.x) || arg4 < fabsf(sp1C.y)) {
+    playerPos.x = player->actor.world.pos.x;
+    playerPos.y = player->actor.world.pos.y + offsetY;
+    playerPos.z = player->actor.world.pos.z;
+
+    func_8002DBD0(&this->dyna.actor, &relPlayerPos, &playerPos);
+
+    if (fabsf(relPlayerPos.x) > maxDistSides || fabsf(relPlayerPos.y) > maxDistY) {
         return FLT_MAX;
     } else {
-        return sp1C.z;
+        return relPlayerPos.z;
     }
 }
 
-s32 func_809968D4(DoorShutter* this, PlayState* play) {
+/**
+ * Returns 0 if the player is not near and facing the door.
+ * Returns 1 if the player is near and faces the front of the door.
+ * Returns -1 if the player is near and faces the back of the door.
+ */
+s32 DoorShutter_GetPlayerSide(DoorShutter* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
     if (!Player_InCsMode(play)) {
-        ShutterInfo* temp_v1 = &sShutterInfo[this->unk_16C];
-        f32 temp_f2 = func_80996840(play, this, (this->unk_16C != 3) ? 0.0f : 80.0f, temp_v1->e, temp_v1->f);
-
-        if (fabsf(temp_f2) < 50.0f) {
-            s16 phi_v0 = player->actor.shape.rot.y - this->dyna.actor.shape.rot.y;
-
-            if (temp_f2 > 0.0f) {
-                phi_v0 = 0x8000 - phi_v0;
+        DoorShutterGfxInfo* gfxInfo = &sGfxInfo[this->gfxType];
+        f32 playerDistForwards =
+            DoorShutter_GetPlayerDistance(play, this, (this->gfxType != DOORSHUTTER_GFX_JABU_JABU) ? 0.0f : 80.0f,
+                                          gfxInfo->rangeSides, gfxInfo->rangeY);
+
+        if (fabsf(playerDistForwards) < 50.0f) {
+            s16 yawDiff = player->actor.shape.rot.y - this->dyna.actor.shape.rot.y;
+
+            // If the player is in front of the door (positive forwards distance),
+            // it is expected that the player faces the opposite way compared to the door.
```

Comment:
"it is expected that the player is facing" imo

Reply:
and maybe we can say "If the player is on the front side" as opposed to "is in front"?
"is in front" to me can be true for either side of the door, whereas if we say player is on the front side we are being clear about the front and back side that each door has

Reply:
Yeah that whole comment applies to both sides of the door
it explains the three-lines if block below

I agree it may not be super clear, suggestions welcome

Reply:
oh ok, I think I took "positive forward distance" to mean that its accounting for the "side". But thats probably more an issue on my part than with how its actually wordedd

Reply:
To explain with more words, in `DoorShutter_GetPlayerSide`

```c
if (fabsf(playerDistForwards) < 50.0f) {
```
checks if the player is near-ish on either side of the door (note `playerDistForwards = DoorShutter_GetPlayerDistance()` returns the max float if player is too far to the side)

```c
s16 yawDiff = player->actor.shape.rot.y - this->dyna.actor.shape.rot.y
```
is the difference in orientation from the door's forwards to the player's (i.e. from the door's forwards to the direction player is looking in)
For example
- if the player is looking at the door from the front side of the door, yawDiff would typically be around half a turn (0x8000), since player looks the other way compared to the door's forwards direction
- if the player is looking at the door from the back side of the door, yawDiff would typically be around 0, since player looks in the same direction as the door's forwards

```c
if (playerDistForwards > 0.0f) {
    yawDiff = 0x8000 - yawDiff;
}
```
if the player is located on the front side of the door, playerDistForwards>0
if the player is located on the back side of the door, playerDistForwards<0

If the player is located on the front side, looking towards the door would make yawDiff in the initial calculation around 0x8000 as stated above, so this brings the value back around 0

```c
if (ABS(yawDiff) < 0x3000) {
```
and the resulting yawDiff is checked to make sure the player isn't facing away more than 3/4th of a quarter of a turn from the door

(I don't know why they specifically did `0x8000 - yawDiff`, `yawDiff +- 0x8000` would have worked as well due to s16 wrapping / the abs right after)

Reply:
I tried to improve this comment